### ISSUE TITLE
Notebook as scripts

### ISSUE THREAD

**Issue by hamelsmu** [fastai/execnb#20]
From @seeM 

<img width="701" alt="image" src="https://user-images.githubusercontent.com/1483922/178166205-56c2883e-072b-4f87-a062-a31eb565dc82.png">

**Comment by seeM** [comment]
Since this is such a thin wrapper, it might be move convenient if this could be achieved by piping `nbprocess_export` to `python`, e.g:
```sh
nbprocess_export —stdout script.ipynb | python -
```
(The `--stdout` option doesn't exist yet.)

**Comment by hamelsmu** [comment]
That seems very doable!

**Comment by dleen** [comment]
I was also thinking about this for a bit and came up with this which involves parametrizing and scripts:

Have a cell in your notebook that essentially is a `docment`:
```
#|params
a:int=1 # This is parameter a
b:str="foo" # This is another parameter...
c:int=my_func() # Call a function to get the default value?
```

You can have a markdown cell before the params which acts as the helpstring for the script e.g.
> "This is the rest of the documentation for the script?"

As usual a cell after the `#|params` cell is injected with the values given by the user.

This notebook could be run via a new function like:
```
def exec_nb_params(
    ...
    *args
    **kwargs
)
```
where `args` and `kwargs` are the positional and optional arguments. The injection
with `inject_idx` etc is taken care of by this function .

Because we are using docments and essentially the `call_parse` mechanism used elsewhere in fastai then we can get a CLI script
```
exec_nb_params my_nb.ipynb --help
```
which would print

> This is the rest of the documentation for the script

**Comment by jph00** [comment]
The cell could just have the docstring in it directly:

```python
#|params
a:int=1 # This is parameter a
b:str="foo" # This is another parameter...
c:int=my_func() # Call a function to get the default value?
"This is the rest of the documentation for the script?"
```

Just to confirm my understanding - the purpose of this is to get a help message (and param error checking) for the "script"? And also to have any values passed at the command line injected into the appropriate cell?

Perhaps you could also specific the cell index of the place where the params will be injected? e.g. to inject into cell 2:

```python
#|params 2
```

**Comment by dleen** [comment]
> Just to confirm my understanding - the purpose of this is to get a help message (and param error checking) for the "script"? And also to have any values passed at the command line injected into the appropriate cell?

Correct

> Perhaps you could also specific the cell index of the place where the params will be injected? e.g. to inject into cell 2:

Makes sense

Mostly just exchanging ideas here, haven't implemented anything yet

**Comment by MichaelJFishmanBA** [comment]
I was just looking for a tool to run a notebook with parameters and save the resulting notebook. [Papermill](https://github.com/nteract/papermill) does this, but it's parameters rely on tagging Jupyter cells, which I can't do easily in VSCode.

**Comment by seeM** [comment]
I don't think it would be too hard to get a working prototype of this with execnb. If anyone would like to take a shot, I'm happy to share pointers. I unfortunately won't be able to get around to it myself just yet
